hash.c

```diff
@@ -6214,6 +6214,33 @@ Init_Hash(void)
     /* Document-class: ENV
      *
      * ENV is a hash-like accessor for environment variables.
+     *
+     * === Interaction with the Operating System
+     * The ENV object interacts with the operating system's environment variables:
+     * - When you get the value for a name in ENV, the value is retrieved from among the current environment variables.
+     * - When you create or set a name-value pair in ENV, the name and value are immediately set in the environment variables.
+     * - When you delete a name-value pair in ENV, it is immediately deleted from the environment variables.
+     * === Names and Values
+     * Generally speaking, each name or value is a String.
+     *
+     * Strictly speaking:
+     * - Each name or value must be one of the following:
+     *   - A String.
+     *   - An object that responds to #to_str by returning a String, which will be used as the name or value.
+     * - A name may not:
```

Comment:
Also a name and value may not contain NUL character.

Reply:
I saw nothing about that in spec.  Will add to spec soon.  And will add to hash.c doc.

Reply:
Added here.

Reply:
`INVALID_ENVVARS` in test/ruby/test_env.rb are examples.

Reply:
Do we need to say more about the encodings?